packages/react/src/button/Button.tsx

```diff
@@ -15,38 +15,70 @@ import { Flex } from "../flex";
 import { Text } from "../text";
 import { type Recipe, recipe } from "./Button.recipe";
 
+const presets = {
+  basic: { colorScheme: "secondary", variant: "solid" },
+  "basic-link": { colorScheme: "secondary", variant: "link" },
+  danger: { colorScheme: "danger", variant: "solid" },
+  "danger-outline": { colorScheme: "danger", variant: "outline" },
+  plain: { colorScheme: "secondary", variant: "ghost" },
+  primary: { colorScheme: "primary", variant: "solid" },
+} satisfies Record<
+  string,
+  { colorScheme: Recipe["colorScheme"]; variant: Recipe["variant"] }
+>;
+
 type ButtonProps = ExtendProps<
   ComponentPropsWithRef<"button">,
   ComponentPropsWithRef<typeof Flex>,
   {
     asChild?: boolean;
     children?: ReactNode;
+    colorScheme?: Recipe["colorScheme"];
+    isLoading?: boolean;
     leftSection?: ReactNode;
+    preset?: keyof typeof presets;
     rightSection?: ReactNode;
+    variant?: Recipe["variant"];
   } & Recipe
 >;
 
 export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
-  ({ asChild, children, leftSection, rightSection, ...props }, ref) => {
+  (
+    {
+      asChild,
+      children,
+      colorScheme,
+      isLoading,
+      leftSection,
+      preset = "basic",
+      rightSection,
+      variant,
+      ...props
+    },
+    ref,
+  ) => {
     const Comp = asChild ? Slot : "button";
     children =
       asChild && isValidElement(children) ? (
         cloneElement(
           children,
           undefined,
-          <Text fontSize="inherit" fontWeight="600">
-            {children.props.children}
-          </Text>,
+          <Text fontSize="inherit">{children.props.children}</Text>,
         )
       ) : (
-        <Text fontSize="inherit" fontWeight="600">
-          {children}
-        </Text>
+        <Text fontSize="inherit">{children}</Text>
       );
 
     const innerRef = useRef<HTMLButtonElement>(null);
     const composedRef = useComposedRefs(ref, innerRef);
 
+    if (isLoading) {
+      props.disabled = true;
```

Comment:
this is not a good pattern - technically `props` is meant to be a read only variable. plus we don't want to forward `disabled` to the element either.

so instead we should extract these 2 props out in the function signature and use them accordingly:

```
const Button = ({ disabled, isLoading, ...props }) => (
  <button data-disabled={disabled || isLoading} {...props} />
)
```